validate/validators/bam.py

```diff
@@ -9,4 +9,8 @@ def validate_bam_file(path):
     except pysam.SamtoolsError as err:
         raise ValueError("samtools bam check failed. " + str(err)) from err
 
+    num_lines = int(pysam.view("-c", str(path)).strip())
+    if num_lines <= 0:
+        raise ValueError("pysam bam check failed. No reads in " + str(path))
```

Comment:
For large BAMs, running the full `samtools view -c` command might take significant time, it would be worth looking into modifying the check to something like `samtools view --no-header <bam> | head -n 1` and checking if what's returned isn't an empty string. Pysam has the [AlignmentFile](https://pysam.readthedocs.io/en/latest/api.html#pysam.AlignmentFile) class, which has a `head` method which might prove useful for doing this.

Reply:
>Pysam has the [AlignmentFile](https://pysam.readthedocs.io/en/latest/api.html#pysam.AlignmentFile) class, which has a head method which might prove useful for doing this.

@aybeshlikyan did you get a chance to try the method? https://pysam.readthedocs.io/en/latest/api.html#pysam.AlignmentFile.head